torch/_inductor/fx_passes/joint_graph.py

```diff
@@ -694,18 +694,34 @@ def pointless_convert(match: Match, arg, dtype1: torch.dtype, dtype2: torch.dtyp
 
 
 def definitely_equal(
-    lhs_sizes: Sequence[Union[torch.SymInt, bool]],
-    rhs_sizes: Sequence[Union[torch.SymInt, bool]],
+    old_sizes: Sequence[Union[torch.SymInt, bool]],
+    new_sizes: Sequence[Union[torch.SymInt, bool]],
 ) -> bool:
     """
     Leverage guard_or_false to compare if two lists of int/symint are equal.
     Useful to compare sizes, strides etc.
+
+    Can handle -1 in new_sizes which happens in the size arguments of a
+    view op.
     """
 
-    return len(lhs_sizes) == len(rhs_sizes) and all(
-        guard_or_false(lhs_item == rhs_item)
-        for lhs_item, rhs_item in zip(lhs_sizes, rhs_sizes)
-    )
+    num_neg1 = 0
+
+    if len(old_sizes) != len(new_sizes):
+        return False
+
+    for lhs_item, rhs_item in zip(old_sizes, new_sizes):
+        if guard_or_false(lhs_item == rhs_item):
```

Comment:
I think the `guard_or_false` calls here are a no-op because:
```py
>>> import sympy
>>> sympy.Symbol("x") == -1
False
>>> sympy.Eq(sympy.Symbol("x"), -1)
Eq(x, -1)
>>> 
```

hash/`==` for sympy expressions are *structural equality* (allowing you to put sympy expressions as dict keys).  It always returns True/False.

Reply:
I see.

I think I could either remove the no-op guard_or_false call, or use sympy.Eq (to be more friendly to dynamic shapes) as your example shows above

Reply:
I was thinking if the size expressions here are sympy expression or torch.SymInt. If they are torch.SymInt maybe behavior is different. I'll try to come up with a test case to verify

Reply:
From the unit test I added, `lhs_item` is a torch.SymInt, `rhs_item` is a fx.Placeholder with torch.SymInt as it's meta['val'].

`lhs_item == rhs_item` or `rhs_item == -1` both being Python True/False. So those guard function calls are no-op.

Reply:
it should not be no ops if you its symInt